Mathlib/Data/ZMod/QuotientRing.lean

```diff
@@ -76,4 +77,24 @@ def ZMod.prodEquivPi {ι : Type*} [Fintype ι] (a : ι → ℕ)
   quotientInfRingEquivPiQuotient _ this |>.trans <|
   RingEquiv.piCongrRight fun i ↦ Int.quotientSpanNatEquivZMod (a i)
 
+/-- The **Chinese remainder theorem**, version for prime factorization. -/
+def ZMod.equivPi_aux :
+    ZMod (∏ (p : n.primeFactors), (p : ℕ) ^ (n.factorization p))
+    ≃+* Π (p : n.primeFactors), ZMod (p ^ (n.factorization p)) :=
```

Comment:
I'm not sure how useful this is once you factor out the lemma that says the prime factors are coprime.

In any case, it shouldn't be an `aux`iliary definition. It should just be inlined into the next one.